Code/Tools/ProjectManager/Source/ProjectExportWorker.cpp

```diff
@@ -0,0 +1,184 @@
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        QFile logFile(GetLogFilePath());
+        if (!logFile.open(QIODevice::WriteOnly | QIODevice::Text | QIODevice::Truncate))
+        {
+            QString error = tr("Failed to open log file.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        EngineInfo engineInfo;
+
+        AZ::Outcome<EngineInfo> engineInfoResult = PythonBindingsInterface::Get()->GetEngineInfo();
+        if (engineInfoResult.IsSuccess())
+        {
+            engineInfo = engineInfoResult.GetValue();
+        }
+        else
+        {
+            QString error = tr("Failed to get engine info.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        QTextStream logStream(&logFile);
+        if (QThread::currentThread()->isInterruptionRequested())
+        {
+            logFile.close();
+            QStringToAZTracePrint(ExportCancelled);
+            return AZ::Failure(ExportCancelled);
+        }
+
+        UpdateProgress(tr("Setting Up Environment"));
+
+        auto currentEnvironmentRequest = ProjectUtils::SetupCommandLineProcessEnvironment();
+        if (!currentEnvironmentRequest.IsSuccess())
+        {
+            QStringToAZTracePrint(currentEnvironmentRequest.GetError());
+            return AZ::Failure(currentEnvironmentRequest.GetError());
+        }
+
+        
+
+        m_exportProjectProcess = new QProcess(this);
+        m_exportProjectProcess->setProcessChannelMode(QProcess::MergedChannels);
+        m_exportProjectProcess->setWorkingDirectory(m_projectInfo.m_path);
+
+        
+        QStringList exportArgs = {"export-project","--project-path", m_projectInfo.m_path};
+        QString commandProgram = QDir(engineInfo.m_path).filePath(GetO3DECLIString());
+
+        m_exportProjectProcess->start(commandProgram, exportArgs);
+        if (!m_exportProjectProcess->waitForStarted())
+        {
+            QString error = tr("Exporting project failed to start.");
+            QStringToAZTracePrint(error);
+            return AZ::Failure(error);
+        }
+
+        while (m_exportProjectProcess->waitForReadyRead(MaxExportTimeMSecs))
+        {
+            QString exportOutput = m_exportProjectProcess->readAllStandardOutput();
+
+            logStream << exportOutput;
+            logStream.flush();
+
+            // Show last line of output
+            if (QStringList strs = exportOutput.split('\n', Qt::SkipEmptyParts);
+                !strs.empty())
+            {
+                UpdateProgress(strs.last());
+            }
+
+            if (QThread::currentThread()->isInterruptionRequested())
+            {
+                // QProcess is unable to kill its child processes so we need to ask the operating system to do that for us
```

Comment:
This comment seems to contradict the code that follows. It looks like we are still trying to kill the process with QProcess.

Reply:
No I think this is talking about something different. The Export process spawns child processes as part of its execution. The code following that comment uses the OS to chase down those processes, kill them, so that the the process can finally terminate without any dangling processes leftover. This logic is also used in the `ProjectBuilderWorker` class